src/cli/README_DATASET.md

```diff
@@ -107,6 +107,108 @@ After the device successfully attaches to a Thread network, the device will retr
+The Leader will initiate the distribution of the Pending Dataset to the rest of the devices in the network.
+
+Normally, an active Commissioner will set a new Pending Dataset. For testing purposes, we will do this in the example
+directly on the Leader using the CLI - so without using a Commissioner.
+
+1. Create a new Dataset in the dataset buffer, by copying the Active Dataset. Then change the channel number.
+
+   ```bash
+   > dataset init active
+   Done
+   > dataset activetimestamp 1696177379
+   Done
+   > dataset pendingtimestamp 1696177379
+   Done
+   > dataset channel 12
+   Done
+   ```
+
+2. Set the delay timer parameter to 5 minutes (300000 ms). Show the resulting Dataset that's ready to be used.
+
+   ```bash
+   > dataset delay 300000
+   Done
+   > dataset
+   Pending Timestamp: 1696177379
+   Active Timestamp: 1696177379
+   Channel: 12
+   Channel Mask: 0x07fff800
+   Delay: 300000
+   Ext PAN ID: 324a71d90cdc8345
+   Mesh Local Prefix: fd7d:da74:df5e:80c::/64
+   Network Key: be768535bac1b8d228960038311d6ca2
+   Network Name: OpenThread-bcaf
+   PAN ID: 0xbcaf
+   PSKc: e79b274ab22414a814ed5cce6a30be67
+   Security Policy: 672 onrc 0
+   Done
+   ```
+
+3. Commit the new Dataset as the Pending Dataset. This also starts the delay timer countdown. The Leader then starts the distribution of the Pending Dataset to other devices in the network.
+
+   ```bash
+   > dataset commit pending
+   Done
+   ```
+
+4. (Optional) Check that the delay timer is counting down, by displaying the Pending Dataset, after a few seconds.
+
+   ```bash
+   > dataset pending
+   Pending Timestamp: 1696177379
+   Active Timestamp: 1696177379
+   Channel: 12
+   Channel Mask: 0x07fff800
+   Delay: 293051
+   Ext PAN ID: 324a71d90cdc8345
+   Mesh Local Prefix: fd7d:da74:df5e:80c::/64
+   Network Key: be768535bac1b8d228960038311d6ca2
+   Network Name: OpenThread-bcaf
+   PAN ID: 0xbcaf
+   PSKc: e79b274ab22414a814ed5cce6a30be67
+   Security Policy: 672 onrc 0
+   Done
+   ```
+
+   This shows that indeed the delay timer has started counting down from its initial value `300000`.
+   The same can be optionally checked on other devices in the network.
+
+
+5. After about 5 minutes, check that the Pending Dataset has been applied at the Leader. This can also be checked at other devices on the network: these should have applied the new Dataset too, at approximately the same time as the Leader has done this.
+
+   ```bash
+   > dataset active
+   Active Timestamp: 1696177379
+   Channel: 12
+   Channel Mask: 0x07fff800
+   Ext PAN ID: 324a71d90cdc8345
+   Mesh Local Prefix: fd7d:da74:df5e:80c::/64
+   Network Key: be768535bac1b8d228960038311d6ca2
+   Network Name: OpenThread-bcaf
+   PAN ID: 0xbcaf
+   PSKc: e79b274ab22414a814ed5cce6a30be67
+   Security Policy: 672 onrc 0
+   Done
+   ```
+
+   This shows that the Active Dataset has now been updated to use channel 12. And the Pending Dataset is no longer present, as can be seen by this command:
```

Comment:
It might be good to have an example of the previous active dataset at the beginning, showing that a different channel was being used?  I.e. a `dataset active` command with the old settings before initializing a new one